test/test_linalg.py

```diff
@@ -180,6 +180,54 @@ def test_det(self, device, dtype):
         with self.assertRaises(RuntimeError):
             op(t)
 
+    @skipCUDAIfNoMagma
+    @skipCPUIfNoLapack
+    @unittest.skipIf(not TEST_NUMPY, "NumPy not found")
+    @dtypes(torch.float32, torch.float64, torch.complex64, torch.complex128)
+    def test_eigh(self, device, dtype):
+        from torch.testing._internal.common_utils import random_hermitian_matrix
+
+        def run_test(shape, batch):
+            matrix = random_hermitian_matrix(shape, *batch, dtype=dtype, device=device)
+            expected_w, expected_v = np.linalg.eigh(matrix.cpu().numpy())
+            actual_w, actual_v = torch.linalg.eigh(matrix)
+            self.assertEqual(actual_w, expected_w)
+            # sign of eigenvectors is not unique and therefore absolute values are compared
```

Comment:
You can make signs deterministic by multiplying each eigenvector with a sing of the largest in absolute real or complex value coordinate, similar to how it is done in `test_lobpcg` in `test_autograd.py`.

Reply:
I could write something like
```python
_, expected_v = np.linalg.eigh(matrix.cpu().numpy())
_, actual_v = torch.linalg.eigh(matrix)
# get the sign for each column
_, idx = actual_v.abs().max(-2, keepdim=True)
sign = actual_v.gather(-2, idx).sign()
is_equal = np.allclose((sign * actual_v).cpu().numpy(), expected_v) or np.allclose(actual_v.cpu().numpy(), expected_v)
self.assertTrue(is_equal)
```
This would require an additional if for a matrix with `(0, 0)` shape, and some tweaking since `.sign()` does not work for complex numbers.
Effectively I think the above code doesn't do better in comparing the numbers to NumPy output than
```python
self.assertTrue(abs(actual_v), abs(expected_v))
```

Reply:
`sing` could be run on either real or complex part. Up to you, just saying, but `abs(actual_v)` will return `|z|`, not `|z| e^{i * \phi}` for each entry, so we are not testing the angular information.

Reply:
I've added the test that compares eigenvectors with the flipped sign, thank you for the suggestion!